### PR TITLE
Add tags support and improve span creation

### PR DESCRIPTION
**Pull request by indrekj**

### CODE REVIEW

File: spec/logasm/tracer_spec.rb [hunk 1]
```diff
@@ -69,6 +69,38 @@
...
+      context 'span context' do
+        it 'has span_id' do
+          expect(span.context.span_id).to_not be_nil
+        end
+
+        it 'has trace_id' do
+          expect(span.context.trace_id).to_not be_nil
+        end
+
+        it 'has does not have parent_id' do
```
Comment (ostap0207) [inline spec/logasm/tracer_spec.rb]: `has does not have`

File: spec/logasm/tracer/span_spec.rb [hunk 2]
```diff
@@ -30,5 +30,18 @@
         }
       )
     end
+
+    context 'when tag set through #set_tag' do
```
Comment (ostap0207) [inline spec/logasm/tracer/span_spec.rb:34]: could be just `#set_tag`

Comment (ostap0207) [reply spec/logasm/tracer/span_spec.rb:34]: oh, it's in the context of `#finish`

**Review by ostap0207 — APPROVED** [review]

File: lib/logasm/tracer.rb [hunk 3]
```diff
@@ -30,7 +30,8 @@ def initialize(logger)
     def start_span(operation_name, child_of: nil, start_time: Time.now, tags: {}, **)
       context =
         if child_of
-          SpanContext.create_from_parent_context(child_of)
+          context = child_of.respond_to?(:context) ? child_of.context : child_of
```
Comment (ostap0207) [inline lib/logasm/tracer.rb]: maybe `parent_context` is better. `context` is used before.

Comment (deiwin) [inline lib/logasm/tracer.rb]: you use `.context`, but documentation mentions `.span_context`

**Review by deiwin — APPROVED** [review]

**Comment by indrekj** [comment]
!squash

File: lib/logasm/tracer.rb [hunk 4]
```diff
@@ -20,9 +20,9 @@ def initialize(logger)
     # Starts a new span.
     #
     # @param operation_name [String] The operation name for the Span
-    # @param child_of [SpanContext] SpanContext that acts as a parent to
+    # @param child_of [SpanContext, Span] SpanContext that acts as a parent to
     #        the newly-started Span. If a Span instance is provided, its
-    #        .span_context is automatically substituted.
+    #        context is automatically substituted.
```
Comment (deiwin) [inline lib/logasm/tracer.rb:25]: Does this mean you're now changing the interface from what OpenTracing defines, though?

Comment (indrekj) [reply lib/logasm/tracer.rb:25]: No https://github.com/opentracing/opentracing-ruby/blob/ddcc0c6ae119b2a95c6ad751f6b9b8116e5f3f11/lib/opentracing/span.rb#L13

Also created https://github.com/opentracing/opentracing-ruby/pull/7

**Comment by deiwin** [comment]
!check

**Pull request merged**